### PR TITLE
Hotfix security api

### PR DESCRIPTION
**Pull request by linuxhunter**
Using C security API snprintf() instead of sprintf()

### CODE REVIEW

File: src/cJSON.c [hunk 1]
```diff
@@ -126,16 +126,16 @@ static char *print_number(cJSON *item)
...
 	{
 		str=(char*)cJSON_malloc(64);	/* This is a nice tradeoff. */
 		if (str)
 		{
-			if (fabs(floor(d)-d)<=DBL_EPSILON && fabs(d)<1.0e60)sprintf(str,"%.0f",d);
-			else if (fabs(d)<1.0e-6 || fabs(d)>1.0e9)			sprintf(str,"%e",d);
-			else												sprintf(str,"%f",d);
+			if (fabs(floor(d)-d)<=DBL_EPSILON && fabs(d)<1.0e60) snprintf(str, sizeof(str), "%.0f",d);
+			else if (fabs(d)<1.0e-6 || fabs(d)>1.0e9)			snprintf(str, sizeof(str), "%e",d);
+			else												snprintf(str, sizeof(str), "%f",d);
```
Comment (CosmosSun) [inline src/cJSON.c]: str is a ptr, sizeof(str) is 8bytes on 64bit machine, it is not corrrect

Comment (linuxhunter) [reply src/cJSON.c]: I have updated the wrong code with digital number 21 and 64 respectively.

File: src/cJSON.c [hunk 2]
```diff
@@ -243,7 +243,7 @@ static char *print_string_ptr(const char *str)
 				case '\n':	*ptr2++='n';	break;
 				case '\r':	*ptr2++='r';	break;
 				case '\t':	*ptr2++='t';	break;
-				default: sprintf(ptr2,"u%04x",token);ptr2+=5;	break;	/* escape and print */
+				default: snprintf(ptr2, sizeof(out)-(ptr2-out), "u%04x",token);ptr2+=5;	break;	/* escape and print */
```
Comment (CosmosSun) [inline src/cJSON.c]: snprintf(ptr2, **(len+3)-(ptr2-out)**, "u%04x",token) maybe better

Comment (linuxhunter) [reply src/cJSON.c]: OK. I have corrected it.

**Pull request merged**